gateway/loaders/zookeeper/utils.go

```diff
@@ -21,50 +21,62 @@ func (z *ZookeeperClient) GetZookeeperData(log zerolog.Logger) (*TargetConfig, e
 	connectionMap := make(map[string]ConnectionConfig)
 	requestMap := make(map[string]RequestConfig)
 
-	zNodes, _, err := z.conn.Children(CleanPath(TargetPath))
-	if err != nil {
-		return nil, err
-	}
-
-	for _, n := range zNodes {
-		c := &ConnectionConfig{}
-
-		crtPath := CleanPath(TargetPath) + CleanPath(n)
-		data, err := z.getNodeValue(crtPath)
+	if exists, _, _ := z.conn.Exists(CleanPath(TargetPath)); exists {
+		zNodes, _, err := z.conn.Children(CleanPath(TargetPath))
 		if err != nil {
 			return nil, err
 		}
 
-		if err := yaml.Unmarshal(data, c); err != nil {
-			log.Error().Err(err).Msgf("Error unmarshalling zNode into yaml: '%s'", crtPath)
-			continue
-		}
+		for _, n := range zNodes {
+			c := &ConnectionConfig{}
 
-		if err := ValidateConnection(c); err != nil {
-			log.Error().Err(err).Msgf("Invalid connection configuration at path: '%s'", crtPath)
-			continue
-		}
+			crtPath := CleanPath(TargetPath) + CleanPath(n)
+			data, err := z.getNodeValue(crtPath)
+			if err != nil {
+				return nil, err
+			}
 
-		connectionMap[n] = *c
-	}
+			if err := yaml.Unmarshal(data, c); err != nil {
+				log.Error().Err(err).Msgf("Error unmarshalling zNode into yaml: '%s'", crtPath)
+				continue
+			}
 
-	zNodes, _, err = z.conn.Children(CleanPath(RequestPath))
-	if err != nil {
-		return nil, err
+			if err := ValidateConnection(c); err != nil {
+				log.Error().Err(err).Msgf("Invalid connection configuration at path: '%s'", crtPath)
+				continue
+			}
+
+			connectionMap[n] = *c
+		}
+	} else {
+		if err := CreateFullPath(z.conn, TargetPath, zk.WorldACL(zk.PermAll)); err != nil {
```

Comment:
It's a bit unusual for a loader/getter to actually create a zk node but it's ok since we were already doing it and we need it in order to setup the zk watchers.